src/components/MainContent/MainContent.tsx

```diff
@@ -0,0 +1,266 @@
+
+  // Todos
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const filteredTodos = (): Todo[] => {
+    switch (currentFilter) {
+      case ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  };
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const isAllTodosCompleted = completedTodos.length === todos.length;
+  const isToggleVisible = completedTodos.length === todos.length
+    && todos.length > 0;
+
+  // Temp todos
+  const [tempTodo, setTempTodo] = useState<TodoForServer | null>();
+
+  const addTempTodo = (title: string) => {
+    const isValidTempTodo = title.trim().length;
+    const isTodoUnique = todos.filter(todo => (
+      todo.title === title
+    )).length === 0;
+
+    if (isValidTempTodo && isTodoUnique) {
+      const newTempTodo = {
+        userId: user.id,
+        title,
+        completed: false,
+      };
+
+      setIsTodoAdding(true);
+      setTempTodo(newTempTodo);
+      setTodos(currentTodos => [...currentTodos, { ...newTempTodo, id: 0 }]);
+      addOnLoadingTodo(0);
+    } else {
+      setError('Can\'t add new todo');
+    }
+  };
+
+  // Server actions
+  const loadTodosFromServer = async () => {
+    try {
+      setTodos(await getTodos(user.id));
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Can\'t load todos');
+    }
+  };
+
+  const addNewTodoOnServer = async (newTodo: TodoForServer) => {
+    try {
+      await addTodo(user.id, newTodo);
+      await loadTodosFromServer();
+      setTempTodo(null);
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to add a todo');
+    }
+  };
+
+  const deleteTodoOnServer = async (id: number) => {
+    try {
+      addOnLoadingTodo(id);
+      await deleteTodo(id);
+      await loadTodosFromServer();
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to delete a todo');
+    }
+  };
+
+  const changeTodoOnServer = async (
+    id: number,
+    todo: Partial<Todo>,
+  ) => {
+    try {
+      addOnLoadingTodo(id);
+      await editTodo(id, todo);
+      await loadTodosFromServer();
+      deleteFromLoadingTodo(id);
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to update a todo');
+    }
+  };
+
+  const changeCompletedForGroup = async (group: Todo[], value: boolean) => {
+    group.map(todo => addOnLoadingTodo(todo.id));
```

Comment:
if you don't need the result of the map method use forEach, that doesn't return nothing